redfish-core/lib/log_services.hpp

```diff
@@ -5249,18 +5249,59 @@ inline void deleteSystemHardwareIsolationLogEntryById(
 
             // Delete the respective dbus entry object
             crow::connections::systemBus->async_method_call(
-                [asyncResp, entryObjPath](const boost::system::error_code ec) {
-                    if (ec)
+                [asyncResp,
+                 entryObjPath](const boost::system::error_code ec,
+                               const sdbusplus::message::message& msg) {
+                    if (!ec)
+                    {
+                        messages::success(asyncResp->res);
+                        return;
+                    }
+
+                    BMCWEB_LOG_ERROR << "DBUS response error [" << ec.value()
+                                     << " : " << ec.message()
+                                     << "] when tried to delete the given "
+                                     << "entry: " << entryObjPath.str;
+
+                    const sd_bus_error* dbusError = msg.get_error();
+
+                    if (dbusError == nullptr)
                     {
-                        BMCWEB_LOG_ERROR << "DBUS response error ["
-                                         << ec.value() << " : " << ec.message()
-                                         << "] when tried to delete the given "
-                                            "object path: "
-                                         << entryObjPath.str;
                         messages::internalError(asyncResp->res);
                         return;
                     }
-                    messages::success(asyncResp->res);
+
+                    BMCWEB_LOG_ERROR << "DBus ErrorName: " << dbusError->name
+                                     << " ErrorMsg: " << dbusError->message;
+
+                    if (strcmp(dbusError->name,
+                               "xyz.openbmc_project.Common.Error."
+                               "NotAllowed") == 0)
+                    {
+                        messages::chassisPowerStateOffRequired(asyncResp->res,
+                                                               "chassis");
+                    }
+                    else if (strcmp(dbusError->name,
+                                    "xyz.openbmc_project.Common.Error."
+                                    "Unavailable") == 0)
+                    {
+                        messages::serviceDisabled(
+                            asyncResp->res, "/redfish/v1/Systems/system/"
+                                            "LogServices/HardwareIsolation");
+                    }
+                    else if (strcmp(dbusError->name,
+                                    "xyz.openbmc_project.Common.Error."
+                                    "InsufficientPermission") == 0)
+                    {
+                        messages::resourceCannotBeDeleted(asyncResp->res);
```

Comment:
This maps to a 403 forbidden here, makes sense 
https://github.com/openbmc/bmcweb/blob/24861a286eb148fcec22afecffb73b596752700f/redfish-core/src/error_messages.cpp#L321

Reply:
Why used different errors ([PropertyValueResourceConflict](https://github.com/ibm-openbmc/bmcweb/pull/309/commits/edb2243cf318414bf0f80a58a12475e95ade25ed#r839847003) and ResourceCannotBeDeleted) for the same backend returned error `InsufficientPermission`?

Because the same request comes from a different URI (one from Core/DIMM to configure and another one coming from DELETE method by using the HardwareIsolation logservices), returning the different error based on the request to give a more appropriate error.

Reply:
> Why used different errors ([PropertyValueResourceConflict](https://github.com/ibm-openbmc/bmcweb/pull/309/commits/edb2243cf318414bf0f80a58a12475e95ade25ed#r839847003) and ResourceCannotBeDeleted) for the same backend returned error InsufficientPermission?

Yes, we should not do this. 
I think this mapping makes sense InsufficientPermission -> resourceCannotBeDeleted; the other one does not

Reply:
I think the context is different from the Redfish URI point of view, right?  i.e one place **the User using the property** (i.e The Enabled property as per Processor schema for the Core and Memory schema for the DIMM) and **the User using the HTTP DELETE request** (i.e DELETE request in the HardwareIsolation LogEntry) to deconfigure it.

Please share your view on this context or You are referring to preferring the 1-to-1 error mapping i.e DBus to Redfish even if the user context (aka URI) is different? IMO, we should give a detailed error message for a better understanding rather than sticking with 1-to-1 mapping.

If you have a different opinion, please share, I like to listen to your thoughts.

Reply:
@gtmills I think PropertyValueResourceConflict error suits to this particular case with respect to use-case.
 it can provide conflicting resource URI to client so that client can understands the error details and take further actions accordingly.

Reply:
Already using the `ResourceCannotBeDeleted` as per request.